The compilation failure is

```
05:59:57,308 INFO  - cd /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/build/Release/src/nvim && /usr/bin/gcc -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -DDISABLE_LOG -I/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/build/Release/config -I/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src -isystem /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps/usr/include -isystem /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps/usr/include/luajit-2.0 -I/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/build/Release/src/nvim/auto -I/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/build/Release/include -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector --param ssp-buffer-size=4 -Werror -Wno-conversion -o CMakeFiles/nvim.dir/eval.c.o -c /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/eval.c
05:59:57,462 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/eval.c: In function ‘f_reltime’:
05:59:57,462 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/eval.c:13217:3: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
05:59:57,462 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/eval.c:13217:3: error: ISO C99 does not support ‘_Static_assert’ [-Werror=edantic]
05:59:57,573 WARN  - cc1: all warnings being treated as errors
```

That happened on neovim-qb-slave-ubuntu-12-04-64bit so unless it is using a locally built compiler, that's likely using

```
$ rmadison -u ubuntu gcc | grep precise
 gcc | 4:4.6.3-1ubuntu5 | precise | amd64, i386
```

--- a/src/nvim/assert.h
+++ b/src/nvim/assert.h
@@ -65,9 +65,16 @@
 # define STATIC_ASSERT_STATEMENT(cond, msg) _Static_assert(cond, msg)
 
 # undef STATIC_ASSERT_PRAGMA_START
+
+#if __GNUC__ >= 6
 # define STATIC_ASSERT_PRAGMA_START \
     _Pragma("GCC diagnostic push") \
-    _Pragma("GCC diagnostic ignored \"-pedantic\"") \
+    _Pragma("GCC diagnostic ignored \"-Wpedantic\"")
+#else
+# define STATIC_ASSERT_PRAGMA_START \
+    _Pragma("GCC diagnostic push") \
+    _Pragma("GCC diagnostic ignored \"-pedantic\"")
+#endif
 
 # undef STATIC_ASSERT_PRAGMA_END
 # define STATIC_ASSERT_PRAGMA_END \

--- a/src/nvim/ex_cmds.c
+++ b/src/nvim/ex_cmds.c
@@ -3816,16 +3816,17 @@ skip:
       EMSG2(_(e_patnotf2), get_search_pat());
   }
 
-  if (do_ask && hasAnyFolding(curwin))
-    /* Cursor position may require updating */
+  if (do_ask && hasAnyFolding(curwin)) {
+    // Cursor position may require updating
     changed_window_setting();
-
-    vim_regfree(regmatch.regprog);
-
-    // Restore the flag values, they can be used for ":&&".
-    do_all = save_do_all;
-    do_ask = save_do_ask;
   }
+
+  vim_regfree(regmatch.regprog);
+
+  // Restore the flag values, they can be used for ":&&".
+  do_all = save_do_all;
+  do_ask = save_do_ask;
+}
 
 /*
  * Give message for number of substitutions.

